mesonbuild/utils/universal.py

```diff
@@ -979,18 +986,74 @@ def default_libdir() -> str:
 
 def default_libexecdir() -> str:
     # There is no way to auto-detect this, so it must be set at build time
+    if is_haiku():
+        return 'lib'
```

Comment:
The comment should be below the Haiku case, since we do auto-detect it on Haiku. :)